File: pkg/reactor/task.go
package reactor

import (
	"time"

	"github.com/fuzzingbits/hub/pkg/entity"
	"github.com/google/uuid"
)

// DatabaseTaskToEntity does what it says
func DatabaseTaskToEntity(dbTask entity.DatabaseTask) entity.Task {
	return entity.Task{
		CanBeCompletedEarly: dbTask.CanBeCompletedEarly,
		Completed:           dbTask.Completed,
		CreatedAt:           dbTask.CreatedAt,
		DeletedAt:           dbTask.DeletedAt,
		DueDate:             dbTask.DueDate,
		Name:                dbTask.Name,
		Notes:               dbTask.Notes,
		UserUUID:            dbTask.UserUUID,
		UUID:                dbTask.UUID,
	}
}

// DatabaseTasksToEntity does what it says
func DatabaseTasksToEntity(dbTasks []entity.DatabaseTask) []entity.Task {
	tasks := []entity.Task{}
	for _, dbTask := range dbTasks {
		tasks = append(tasks, DatabaseTaskToEntity(dbTask))
	}

	return tasks
}

// TaskCreateRequestToDatabaseTask does what it says
func TaskCreateRequestToDatabaseTask(request entity.TaskCreateRequest, loggedInUser entity.UserContext) entity.DatabaseTask {
	return entity.DatabaseTask{
		CanBeCompletedEarly: request.CanBeCompletedEarly,
		Completed:           false,
		CreatedAt:           time.Now(),
		DeletedAt:           nil,
		DueDate:             request.DueDate,
		Name:                request.Name,
		Notes:               request.Notes,
		UserUUID:            loggedInUser.User.UUID,
		UUID:                uuid.New().String(),
	}
}

File: ui/store/task.ts
import * as types from "~/ui/assets/types";

export const state = () => ({
	tasks: [
		{
			uuid: "234-432-432-432",
			name: "Laundry",
			dueDate: "2020-05-22",
			canBeCompletedEarly: true,
			completed: false,
			deletedAt: null,
			notes: "Just a simple note.",
			userUUID: "bla-bla-bla",
		},
	] as types.Task[] | null,
});

export const mutations = {
	setTasks(state: any, target: types.Task[] | null) {
		state.tasks = target;
	},
};
